### ISSUE TITLE
Documentation request: Where did you install the files?

### ISSUE THREAD

**Issue by mcclure** [microsoft/vcpkg#34125]
Today I installed vcpkg. I followed the [getting started](https://vcpkg.io/en/getting-started) instructions and then installed a single package (libusb). It installed many dependencies, then libusb. Great.

…where are the files?

I understand there is a cmake workflow for hooking up to vcpkg. But I wanted to know where the so-installed files *were*, for two reasons; one, I might want to delete them later, because some of the dependency packages installed (Python, CMake, Meson) sounded kind of large; two, I am not using cmake, I am using Rust "cargo", and Rust "cargo" depends on a environment variable named VCPKG_ROOT (this is its own problem, [see here](https://github.com/microsoft/vcpkg/discussions/24239)).

It turns out that the files are installed into the git checkout directory created in the getting started instructions. This git checkout directory also turns out to be the VCPKG_ROOT Rust wants (as identified by a file .vcpkg_root at the top level). I had to figure this out myself through detective work. It isn't described in the Windows quick start guide in the README, or the "Installing and Using Packages" guide the README recommends you read next. The quick start guide even makes reference to "[path to vcpkg]", but does not indicate *what* the path to vcpkg is (the git checkout). Writing this now I *do* find the "Getting Started guide" linked on the separate website vcpkg.io has the sentence "The repo can be cloned anywhere, and will include the vcpkg binary after bootstrapping as well as any libraries that are installed from the command line" but I didn't see this at the time and I personally find this sentence slightly ambiguous.

I guess it was "obvious" to whoever wrote this documentation that the `vcpkg-bootstrap` and `vcpkg install` installs itself *in-place* into the directory vcpkg has been checked out into but it was not obvious to me :( 

**"Expected behavior"**

I recommend explaining this (that the "install directory" is the same as the git checkout directory) more explicitly in the README and also taking note of the [support thread linked above](https://github.com/microsoft/vcpkg/discussions/24239).

Surfacing a little higher in the documentation a link to the explanation of the layout of this magic directory (the significance of .vcpkg_root; which subdirectories are installed packages and which are vcpkg plumbing; whatever it is you do to "reset" the git repo to its uninstalled state, I guess `git clean -xffd`?) would probably be good but is probably not required.

**Environment**

Windows 10 , 10.0.19045 Build 19045
Rust/Cargo 1.69.0
vcpkg… 768619a, I guess?

**Comment by autoantwort** [comment]
Hm if you open the documentation (https://learn.microsoft.com/en-us/vcpkg/) there is `Integrate with a custom buildsystem` (https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/manual-integration). 

Maybe the README.md should link to documentation after the "How to setup"

**Comment by mcclure** [comment]
> Hm if you open the documentation (https://learn.microsoft.com/en-us/vcpkg/) there is `Integrate with a custom buildsystem` (https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/manual-integration).
> 
> Maybe the README.md should link to documentation after the "How to setup"

Hi, thanks for linking me this— this document is great and answers my remaining questions. Two notes, one, the "magic" `.vcpkg_root` file is not documented here, two, this document makes reference to the vcpkg root without explaining what the vcpkg root is. Maybe the "vcpkg root" does not need to be defined in the "custom buildsystem" document if it is defined more clearly elsewhere though.

**Comment by Osyotr** [comment]
vcpkg.io should redirect to learn.microsoft.com/vcpkg IMHO. The latter should also have a section "How it works" or something.

> without explaining what the vcpkg root is

`meow root` is a common programming term describing a folder where `meow` is located.
Now that you know what vcpkg root is, how would you document it?

**Comment by mcclure** [comment]
I agree that "meow root" is clearly "the folder where meow is located", but it is not at all obvious to me that "the folder where meow is located" is the git checkout of its source. My first minutes with vcpkg involved an install step through a "bootstrap" script, followed by a step using the actual name "install" step. Often, pieces of software which are distributed as software and have "install" steps such as this install to some *other* location (such as ~/.something or %APPDATA%), after which the distribution/source repo can be deleted. Although the "in-place" nature of the vcpkg repo/distribution is now clear to me, it was not inherently obvious at first.

> Now that you know what vcpkg root is, how would you document it?

The getting started guide in the README begins with

> First, download and bootstrap vcpkg itself; it can be installed anywhere, but generally we recommend using vcpkg as a submodule so the consuming repo can stay self-contained. Alternatively, vcpkg can be installed globally; we recommend somewhere like C:\src\vcpkg or C:\dev\vcpkg, since otherwise you may run into path issues for some port build systems.

… followed by sample invocations of git and `.\vcpkg\bootstrap-vcpkg.bat`. I would reword this as:

> First, download and bootstrap vcpkg itself:
>
>> Sample code
>
> The directory from which `bootstrap-vcpkg` was run is now a functioning vcpkg installation, and all packages downloaded will be installed within it. This directory can be installed anywhere, but generally we recommend using vcpkg as a submodule so the consuming repo can stay self-contained. Alternatively, vcpkg can be installed globally; we recommend somewhere like C:\src\vcpkg or C:\dev\vcpkg, since otherwise you may run into path issues for some port build systems.

Alternately, if the "root" language is important vocabulary to vcpkg (and not just an affectation of Cargo) the added sentence could refer to "a functioning vcpkg root" instead of "a functioning vcpkg installation.

Additionally, I would add a sentence to the getting started guide clarifying under what circumstances it is allowed to move the vcpkg root (for example: do things break if you move the directory after bootstrapping? if so, say so after you describe how to bootstrap. do things break if you move the directory after running `integrate`? if so, say so after you describe how to integrate).

**Comment by SchaichAlonso** [comment]
> Often, pieces of software which are distributed as source and have "install" steps such as this install to some _other_ location (such as ~/.something or %APPDATA%), after which the distribution/source repo can be deleted.

VCPKG defaults to [binary caching](https://learn.microsoft.com/en-us/vcpkg/users/binarycaching) using the file provider, which results in each installed package being archived into a zip file which is stored in %APPDATA% . A second attempt to install the same package set (with the same vcpkg revision on the same system) will simply extract those zip files, while build-time-only dependencies won't be installed (unless you explicitly ask for them).

If you do not "need" vcpkg after it has provided for whatever libraries you wanted it to provision, you can use `vcpkg export` to have vcpkg generate a bundle with your libraries and their runtime (but not compile-time) dependencies, and once you have that bundle, delete the vcpkg checkout.

**Comment by jimwang118** [comment]
We hope your question was answered to your satisfaction; if it wasn't, you can reopen with more info.
